### PR TITLE
Add schema validation on the configuration file

### PR DESCRIPTION
**Pull request by mujx**

### CODE REVIEW

File: src/kompos/cli/terraform.py [hunk 1]
```diff
@@ -182,12 +182,6 @@ def __init__(self, root_dir, cluster_config_path, kompos_config, execute):
 
     def run(self, args, extra_args):
         logger.info("Found extra_args %s", extra_args)
-        terraform_config_path = os.environ.get(
-                                                "TF_CLI_CONFIG_FILE",
-                                                self.kompos_config.terraform_config_path
-                                                )
-        os.environ["TF_CLI_CONFIG_FILE"] = terraform_config_path
```
Comment (mujx) [inline src/kompos/cli/terraform.py:189]: Since this option is unrelated to kompos, I removed it from the config.

File: src/kompos/komposconfig.py [hunk 2]
```diff
@@ -38,72 +42,24 @@ def get_value_or(dictionary, x_path, default=None):
...
-    For /root/cluster/cluster.yaml:
-
-        /etc/kompos/.komposconfig.yaml
-        ~/.komposconfig.yaml
-        /root/.komposconfig.yaml
-        /root/cluster/.komposconfig.yaml
+    Parses all the available configuration files in order and merges them together.
     """
 
-    DEFAULTS = {
```
Comment (mujx) [inline src/kompos/komposconfig.py:74]: Default values can be placed directly in the schema definition.

File: src/kompos/komposconfig.py [hunk 3]
```diff
@@ -38,72 +42,24 @@ def get_value_or(dictionary, x_path, default=None):
...
-
-        # S3 remote state
-        'terraform.s3_state': False,
-
-        # Integrate https://github.com/coinbase/terraform-landscape
-        'terraform.landscape': False,
-
-        # Remove .terraform folder before each terraform plan, to prevent reuse of installed backends (it can confuse terraform when the cluster backend is
-        # not the same for all of them)
-        'terraform.remove_local_cache': False,
```
Comment (danielcoman) [inline src/kompos/komposconfig.py:89]: I think this needs to be enforced 'true' and not make it an option.

**Review by danielcoman — APPROVED** [review]

File: src/kompos/komposconfig.py (hunk 3 above)
Comment (mujx) [reply src/kompos/komposconfig.py:89]: Yes it's done by default.
